test/sharness/t0080-repo.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/sh
+#
+# Copyright (c) 2014 Jeromy Johnson
+# MIT Licensed; see the LICENSE file in this repository.
+#
+
+test_description="Test ipfs repo operations"
+
+. lib/test-lib.sh
+
+test_init_ipfs
+test_launch_ipfs_daemon
+
+test_expect_success "'ipfs add afile' succeeds" '
+	echo "some text" > afile
+	HASH=`ipfs add -q afile`
+	echo -n $HASH > hashfile
+'
+
+test_expect_success "added file was pinned" '
+	ipfs pin ls -type=recursive | grep $HASH
+'
+
+test_expect_success "'ipfs repo gc' doesnt remove file" '
+	echo -n "" > empty
+	ipfs repo gc > gc_out_actual
+	test_cmp empty gc_out_actual
+	ipfs cat $HASH > out
+	test_cmp out afile
+'
```

Comment:
i think we have to split these, or a failure will be hard to debug:

```
test_expect_success "'ipfs repo gc' succeeds" '
 ipfs repo gc > gc_out_actual
'

test_expect_success "'ipfs repo gc' looks good (empty)" '
 echo -n "" > empty &&
 test_cmp empty gc_out_actual
'

test_expect_success "'ipfs repo gc' doesnt remove file" '
 ipfs cat $HASH > out &&
 test_cmp out afile
'
```

also, i think we need the `&&` between the commands for it to work as expected